src/privacy_analysis/packet_analysis/packet_privacy_port.py

```diff
@@ -1,44 +1,75 @@
+        self.packet = packet_input
+        is_TCP = self.packet.haslayer(TCP)
+        is_UDP = self.packet.haslayer(UDP)
 
         # Perform TCP and UDP checks
         if is_TCP or is_UDP:
             proto_type = TCP if is_TCP else UDP
 
             # Scan for using port 80 and the plaintext for privacy leaks
-            if (packet[proto_type].dport == 80) or (packet[proto_type].sport == 80):
+            if (self.packet[proto_type].dport == 80) or (self.packet[proto_type].sport == 80):
                 alert_port_80 = Alert(None, "Sending data over unencrypted port.", ALERT_TYPE.PRIVACY, SEVERITY.ALERT)
                 alert_port_80.alert()
-                self.__scan_plaintext(packet, proto_type)
+                self.__scan_plaintext(proto_type)
 
             # Monitor suspicious ports
-            print("Monitoring suspicious ports")
-            if packet[proto_type].dport in suspicious_ports:
-                alert_suspicious_ports = Alert(None, "Suspicious destination port used: " + packet[proto_type].dport,
-                                               ALERT_TYPE.PRIVACY, SEVERITY.WARN)
+            if self.packet[proto_type].dport in suspicious_ports:
+                alert_suspicious_ports = Alert(None, "Suspicious destination port used: " +
+                                               str(self.packet[proto_type].dport), ALERT_TYPE.PRIVACY, SEVERITY.WARN)
                 alert_suspicious_ports.alert()
-                print("Alert on bad port")
 
     # Scan the plaintext for privacy leaks
-    # TODO: regex for email, SSN, credit card
-    def __scan_plaintext(self, packet, proto_type):
+    def __scan_plaintext(self, proto_type):
+        # Try to get the payload
         try:
-            if (packet[proto_type].dport == 80) or (packet[proto_type].sport == 80):
-                print(packet[proto_type].payload)
+            self.payload = str(self.packet[proto_type].payload)
         except:
-            print("plaintext not found")
-            print(packet[proto_type].payload)
             return
+
+        # TODO: There is a huge chance for false positives with credit cards and SSNs, because they're just numbers
+        # TODO: Should we include them?
+        # Use a regex to look for credit cards, needs a non-number before and after the numbers
+        self.__regex_alert("[\D](?:[0-9]{4}-){3}[0-9]{4}[\D]|[\D][0-9]{16}[\D]", "Credit card information found in a plaintext packet.")
+
+        # Use a regex to look for SSNs, needs a non-number before and after the numbers
+        self.__regex_alert("[\D][0-9]{9}[\D]|[\D][0-9]{3}-[0-9]{2}-[0-9]{4}[\D]", "SSN information found in a plaintext packet.")
+
+        # Use a regex to look for emails - this is not a huge privacy leak, but still mentionable
+        # The email will not be included in the alert for privacy reasons
+        self.__regex_alert("[^@]+@[^@]+\.[^@]+", "Email information found in a plaintext packet.")
```

Comment:
Not too much of a concern, but this could trigger as a false positive on the raw data kinda commonly. '@' is kinda common in some of the raw stings when not as bytes.

Reply:
Yeah I found a bunch of long and complicated regexes, all claiming to be the messiah of email regexes, and this was the only one that worked for my input data lol. I could try the long and complicated ones on simpler emails for the test data, but if we wanna get one consistently this would be better. It's just a balance of false positives vs false negatives